### DISCUSSION TITLE
K1 Max Camera not working

### DISCUSSION THREAD

**Discussion by pdsavas** [Guilouz/Creality-Helper-Script-Wiki#621] [Issues]
Hi,

I recently restored to factory settings in order to update to the latest version of the firmware on my Max. Before I rooted the printer, and went through the process of installing Mainsail, etc. on the printer, I noticed the Webcam was working in the Creality interface (which I have removed with the Helper Script). Unfortunately when I got into Mainsail, all I see is a white screen under the Webcam section on the dashboard. Similarly, when I go into the settings and select the Webcam, it shows a camera is there at the printer's IP, but does not permit editing that camera.

![image](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/46990222/494aa9f4-d7a6-4427-9c3f-4c476c40592a)

I can add another camera, but still do not get a feed from it when using the default settings. Under the MPEG-Streamer option, I get the white rectangle; if I try the Adaptive option (or any other of the options) it's just black.

![image](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/46990222/14301eca-2cf1-473e-adcc-997051028563)

I did try adding "http:/xxx.xxx.xxx.xxx/" in front of "/webcam/?action=stream" and "/webcam/?action=snapshot" (a possible solution I saw in another Fluidd-related thread), but that didn't make a difference.

I did notice Crowsnest is not installed, nor is there an option to install it from the Helper Script; could this be the issue?

How can I get the camera working?

Any help would be appreciated.

Thanks,
pete

**Comment by Uplinkpro** [comment]
Im having this exact issue on mine, did you ever get this fixed?

**Comment by tylamp** [comment]
Same issue

**Reply by Uplinkpro** [reply]
Found out on mine that there was something that creality broke with the latest firmware with the camera, theres a script for to fix it

**Reply by tylamp** [reply]
Where can I find the script?

**Reply by mklengel** [reply]
Same on my K1C with the latest firmware. The web version of Creality Print doesnt show the camera stream.

**Reply by Uplinkpro** [reply]
### 🛠️ Manual Camera Fix for Creality K1 Max (Firmware 1.3.5.22+)

Recent firmware updates changed how the camera stream is handled, routing it through an internal app to a shared memory buffer rather than a standard video device. This breaks older Crowsnest/Mainsail/Fluidd configurations.

Follow these steps to manually update your configuration to point to the new memory buffer pipeline.

#### 1. Connect to your printer via SSH
Log in as `root` using your preferred SSH terminal.

#### 2. Edit your Crowsnest Configuration
Open your `crowsnest.conf` file:
```bash
vi /usr/data/crowsnest/crowsnest.conf
```

#### 3. Update the Webcam Device Settings
Locate your active `[webcam]` configuration block and update the parameters to match the following. The critical change is updating the `device` path to point to the shared memory location:

```ini
[webcam device]
mode: ustreamer
port: 8080
device: /dev/shm/camera_main   # Points to Creality's updated shared memory pipe
resolution: 1280x720
max_fps: 15
v4l2ctl: focus_auto=0          # Disables auto-focus loops introduced by the firmware
```

#### 4. Restart the Service
Apply the changes by restarting the camera service daemon:
```bash
/etc/init.d/S50crowsnest restart
```

**Reply by mklengel** [reply]
Thx for your response. In my case (K1C) there is no /usr/data/crowsnest folder.

```
root@K1C-D9EF /root [#] ls /usr/data/
ai_image                 helper-script            machine_production_info  nginx                    swap
creality                 lost+found               mainsail                 opt                      timelapse
fluidd                   macaddr.txt              moonraker                printer_data             wpa_supplicant.conf
root@K1C-D9EF /root [#]
```

**Reply by rodtin** [reply]
For my K1MAX I could not find crowsnest folder as well
